### PR TITLE
eng-206

### PR DESCRIPTION
**Pull request by Connorbelez**
# ENG-206: Disbursement Bridge Implementation

This pull request implements the disbursement bridge that converts eligible dispersal entries into outbound lender payout transfers, completing the lender disbursement flow.

## Core Features

- **Bridge Module**: New `disbursementBridge.ts` with batch processing, per-entry validation, and idempotency
- **Transfer Effects Integration**: Dispersal entry status lifecycle tied to transfer confirmation/failure
- **Daily Alert Cron**: Monitors pending entries past hold period for admin visibility
- **Comprehensive Testing**: Unit and integration tests covering happy path, edge cases, and failure scenarios

### Bridge Functions
- `triggerDisbursementBridge`: Batch orchestrator finding eligible entries and creating transfers
- `processSingleDisbursement`: Per-entry processing with disbursement gate validation
- `findEligibleEntriesInternal`: Query for pending entries past hold period
- `resetFailedEntry`: Recovery mechanism for failed disbursements
- `checkDisbursementsDue`: Daily cron alert handler

### Transfer Lifecycle Integration
- `publishTransferConfirmed`: Updates dispersal entry to "disbursed" with payout date
- `publishTransferFailed`: Marks dispersal entry as "failed" for retry
- `publishTransferReversed`: Handles disbursement reversals

### Pipeline Support
- Multi-leg deal closing pipeline with Leg 1 (buyer → trust) and Leg 2 (trust → seller)
- Pipeline status computation from transfer leg statuses
- Automatic Leg 2 creation when Leg 1 confirms

## Technical Details

- **Idempotency**: `disbursement:{dispersalEntryId}` keys prevent duplicate transfers
- **Amount Preservation**: Uses `entry.amount` as-is per ENG-219 (no recomputation)
- **Balance Gate**: Validates disbursements against LENDER_PAYABLE balance
- **Provider Support**: Phase 1 uses `mock_eft`, ready for VoPay integration
- **Error Handling**: Graceful failure with detailed error reporting

## Testing Coverage

- Unit tests for helper functions and idempotency
- Integration tests for full disbursement lifecycle
- Edge case coverage including concurrent runs, insufficient balance, and transfer failures
- ENG-219 guard test ensuring amount passthrough

## Cron Schedule Update

Added daily disbursement check at 09:00 UTC (after payout batch at 08:00) for admin alerts.

## ENG-217: Servicing Fee Model Documentation

Enhanced documentation for the servicing fee calculation model, confirming current implementation decisions.

## Key Clarifications

- **Fee Basis**: Current outstanding principal (`mortgage.principal`) at settlement time
- **Fee Timing**: Pre-disbursement deduction during dispersal entry creation
- **Principal Sensitivity**: Fees decrease proportionally as principal is repaid

## Documentation Updates

- Added comprehensive JSDoc to `calculateServicingFee` explaining principal basis
- Inline comments in `calculateServicingSplit` documenting ENG-217 decisions
- New test case verifying fee decreases when mortgage principal decreases

## Technical Implementation

- Fee calculated using current outstanding balance, not original loan amount
- `principalBalance` stored in `servicingFeeEntries` for audit verification
- Standard amortizing mortgage behavior where servicing fees decrease with principal paydown

This addresses Tech Design §10 Open Decision 2 and Foot Gun 7, confirming the current implementation follows standard mortgage servicing practices.

## Release Notes

* **New Features**
  * Introduced automated disbursement bridge to batch process eligible loan disbursements efficiently
  * Added daily automated checks to identify disbursements ready for processing
  * Enhanced tracking of disbursement status throughout the payout lifecycle

* **Bug Fixes**
  * Improved error handling and retry capability for failed disbursement operations

* **Tests**
  * Added comprehensive test coverage for disbursement workflows and edge cases

### CODE REVIEW

File: specs/ENG-206/eng-206-disbursement-bridge-plan.html [hunk 1]
```diff
@@ -0,0 +1,1283 @@
...
+		.badge-sky {
+			color: var(--text-inverse);
+			background: var(--sky);
+		}
+		</style>
+	</head>
+	<body>
+		<button type="button" class="theme-toggle" onclick="toggleTheme()">
+			◐ theme
+		</button>
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply specs/ENG-206/eng-206-disbursement-bridge-plan.html]: **Fix:** Exposed the underscore-prefixed functions as properly-named globals so all buttons work correctly.

```js
window.toggleTheme = _toggleTheme;
window.mermaidZoom = _mermaidZoom;
window.toggleExpand = _toggleExpand;
```

All button `onclick` attributes now resolve correctly.
